test/catch2/README.md

```diff
@@ -0,0 +1,39 @@
+# WiredTiger catch2 framework tests
```

Comment:
When using 'Catch2' as a word describing the unit test framework, the correct spelled is with a capital 'C' (see https://github.com/catchorg/Catch2)

Reply:
I also suggest adding a sentence here that says what Catch2 is and linking to https://github.com/catchorg/Catch2 .